### DISCUSSION TITLE
Discussions about Open Voice Interoperability documents and specifications

### DISCUSSION THREAD

**Discussion by dadahl** [open-voice-interoperability/docs#2] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community about our specifications and documentation. We hope that you:
  * Ask questions you’re wondering about.
  * Offer suggestions for improving our documents and specifications.
  * Suggest any new documents that you would find helpful.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together.

**Comment by TheBest-Name** [comment]
I was just reading your web page and at first I thought I knew what this was supposed to be but now I'm not sure I thought it would be a way to hook up assistance like Alexa to be able to use something like Google home and other assistants like that but now I'm wondering is it just for agents like the ones that AI models can fork off to do other tasks 

Also I'm wondering what is the difference between this and MCP server because it almost seems like this is an MCP server for agents? 

But thank you for reading this.

**Comment by dadahl** [comment]
Hi,

Thanks for your note. 

The Open Floor Protocol is a message format that enables agents to communicate with each other. In theory, it could be used for assistants like Google Home and Echo to interoperate, but there would have to be some kind of middleware or adapter between their native protocols and OFP. But you’re right that it can be used for conversational agents to pass task along to other agents.  That’s an interesting idea about this being like an MCP server for agents. I haven’t thought about that, but it kind of makes sense. Usually MCP servers aren’t agents, according to my understanding, but there’s no reason they couldn’t be. 

The more recent ideas we’re exploring are looking at a group of agents all gathered together in a conversational floor where they all contribute ideas to the discussion.

If you’re interested, we welcome new contributors.

**Reply by TheBest-Name** [reply]
Okay, thank you. 

To be honest MCP was the first thing I thought of because you have the from what I could tell from your protocol the origination point that then sends a message to an appropriate endpoint based on a description that is provided to the originator and then when the endpoint whether that be a tool or an agent gets the message they act upon it do whatever they're going to do and then they return a message that the originator can then act on and display or do whatever it's going to do and you can chain tool calls or I guess in this case agent calls.

Either way this seems like a really cool idea and I'm just wondering would that be possible to combine the two or is that possible that one agent could share its tool calling abilities as well?

Is it also possible to get multiple opinions as in you could send the same message to two different agents to compare results? 

Thank you again.

**Reply by TheBest-Name** [reply]
I did also want to add that one idea for what you said about having all the agents gathered to talk to each other, is that there should be a way for the human element or the initiator whoever that be (I guess human or AI) to interject and add stuff in the middle of the conversation.
That way the other participants don't start going off on a tangent so to speak and you can wrangle them back into where you want them to start talking, because I was just trying out AI from co-pilot that had a bunch of AIs talking together and they kept going off on tangents because I couldn't route them in the way that I wanted to and they kept starting to want to do their own thing

**Comment by dadahl** [comment]
Yes, you are absolutely correct, that is an excellent point. I have also seen that happen with multiple agents on a floor. In the OFP floor, the user is most often present, although they don’t have to be. Besides wrangling the agents when they go off on tangents, the other reasons for having the user in the conversation are for the user to be able to interject new information and react to what the  agents are saying. We also have the concept of an agent that manages the agents on the floor, including managing unruly agents, called the “convener” . We’re still developing convener agents, so we don’t have any examples just yet.

Here are a couple of transcripts of conversations with agents and a user. One is a user with two agents that disagree on financial advice, and the other is a user with an agent that always delivers hallucinations (erin) and a hallucination checker (verity).

**Reply by TheBest-Name** [reply]
Yeah I heard about the conveyor, speaking of that, does it allow connection between two assistants one of which isn't the initiator?

I didn't see the transcripts I assume there was supposed to be a link but if one is hallucinating and the other one verifies hallucinations unless you say explicitly that one is the source of truth on a matter how does one know when another is saying something wrong or that it's itself is saying something wrong I would feel like the best way to do that would be by majority rule but in a matter of only two then you have to decide by saying which would be the most correct on a certain topic. 

One more thing is how does the conveyor know when it is getting off topic? Is there some sort of check and balance system where another agent can tell it if it is going off topic because with enough messages from other agents it might start to lean towards them I would think?

**Comment by dadahl** [comment]
Here’s a link to the transcripts of agents offering opposing advice and the hallucination checker identifying hallucinations  https://github.com/open-voice-interoperability/implementation-examples/blob/main/agentTranscripts.pdfThese are great questions. Thanks for raising these topics. The default is that all of the agents present on the floor will be connected to each other, including the convener (not conveyer). Agents can send private messages to each other if they want to, but normally all the agents hear all the messages, so yes, any assistants can be connected.That’s an interesting question about who gets to decide if an agent is hallucinating. The Verity agent in the transcript is designed for the sole purpose of checking for hallucinations, but of course it is based on an LLM itself, and could be hallucinating. Often a user will be present in the conversation who can use their judgement about whether something that is flagged as a hallucination is really a hallucination, but obviously that isn’t foolproof either. If someone wanted to implement a majority rule process, that would be possible by including several hallucination checkers in a floor, which would send their opinions to the convener and/or user to make a final determination.  We’ve written a couple of papers about our hallucination checking process, and we’ve gotten good results in our experiments. Here’s a paper on our experiments https://arxiv.org/abs/2501.13946How the convener can tell if an agent is getting off-topic is a very active area of investigation.  Or more generally, if an agent is getting “unruly” (repeats itself, or comes up with a lot of hallucinations for example) and needs to have the floor revoked or be uninvited.  We’re looking at both LLM-based and rule-based approaches to this and maybe a combination of LLMs and rules would end up working well. That being said, detecting unruly behavior isn’t strictly speaking part of the specification, because agents (including the convener) are black boxes, but we want to understand how a convener could work.How the convener would know which agent would be the most correct on a certain topic could be based on looking at the manifests of the agents and seeing what their expertise is. For example, a convener might trust Stella, an astronomy expert,  for astronomy facts but not Prudence, which is a financial expert. It would be interesting to look into that.The source for Verity, Prudence and Stella, along with other agents  can be found in the “implementation-examples” repository, if you’re interested.From: TheBest_Name! ***@***.***>Date: Friday, May 15, 2026 at 11:03 AMTo: open-voice-interoperability/openfloor-docs ***@***.***>Cc: Deborah Dahl ***@***.***>; Author ***@***.***>Subject: Re: [open-voice-interoperability/openfloor-docs] Discussions about Open Voice Interoperability documents and specifications (Discussion #2)Yeah I heard about the conveyor, speaking of that, does it allow connection between two assistants one of which isn't the initiator?I didn't see the transcripts I assume there was supposed to be a link but if one is hallucinating and the other one verifies hallucinations unless you say explicitly that one is the source of truth on a matter how does one know when another is saying something wrong or that it's itself is saying something wrong I would feel like the best way to do that would be by majority rule but in a matter of only two then you have to decide by saying which would be the most correct on a certain topic.One more thing is how does the conveyor know when it is getting off topic? Is there some sort of check and balance system where another agent can tell it if it is going off topic because with enough messages from other agents it might start to lean towards them I would think?—Reply to this email directly, view it on GitHub, or unsubscribe.Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by TheBest-Name** [reply]
I was just thinking and what if you had a majority vote system where instead of a Verity machine checking what is right what is wrong what about whenever a agent says something that needs to be checked it is sent to all other agents with a simple is this correct? Yes, no, I don't know answer. Then you treat the answer as a one and the other two a zero and then send that to a Verity agent that looks over the manifest and provides a weighted value based on if it thinks the agent would have expertise in the field. Based off that it wouldn't be a sign of value between 0 and 1 based on what the expertise would be. And of course at that point it would just be a weighted average. I don't know if you would just want to throw away the "I don't know"s or if you would want to keep them but either way you would have a weighted average to say more yes or more no, or if it was in the middle then you could just say that it was uncertain either way.
That way you can have majority rule while still keeping some assemblance of expertise matters.

**Reply by TheBest-Name** [reply]
My previous comment about a majority rule idea taking into account expertise of different models, do you think that is going in the right direction or the wrong direction given what you had said previously about It would be interesting to take into account the model looking at which agent had more experience?

Given that certain models can error, it seems like it might be best to have majority be the main factor, and less so an overarching model, because if the overarching model errors, then it might be wrong even if the models underneath aren't.
